userdetails-plugin/src/main/groovy/au/org/ala/userdetails/IUserService.groovy

```diff
@@ -71,13 +73,13 @@ interface IUserService {
 
     List<String[]> countByProfileAttribute(String s, Date date, Locale locale)
 
-    void findScrollableUsersByUserName(String username, int maxResults, ResultStreamer resultStreamer)
+    void findScrollableUsersByUserName(GrailsParameterMap params, ResultStreamer resultStreamer)
 
-    void findScrollableUsersByIdsAndRole(List<String> ids, String roleName, ResultStreamer resultStreamer)
+    void findScrollableUsersByIdsAndRole(GrailsParameterMap params, ResultStreamer resultStreamer)
 
     def getUserDetailsFromIdList(List idList)
 
-    UserRecord findByUserNameOrEmail(String username)
+    UserRecord findByUserNameOrEmail(GrailsParameterMap params)
```

Comment:
What's the thinking behind replacing the parameters here with the GrailsParameterMap map?

Reply:
I had to change it because Cognito findByUserNameOrEmail method is using listUsers(GrailsParameterMap params) to search the users. I could not find a way to create a GrailsParameterMap in CognitoUserService. Is there a way to da that?

Reply:
Talked through with Yasima offline, GPM use is the lesser evil, the alternative is listing all parameters required by every implementation.